priam/src/main/java/com/netflix/priam/cluster/management/IClusterManagement.java

```diff
@@ -15,12 +15,72 @@
  */
 package com.netflix.priam.cluster.management;
 
+import com.netflix.priam.IConfiguration;
+import com.netflix.priam.merics.IMeasurement;
+import com.netflix.priam.merics.IMetricPublisher;
+import com.netflix.priam.scheduler.Task;
+import com.netflix.priam.scheduler.TaskTimer;
+import com.netflix.priam.utils.JMXConnectionException;
+import com.netflix.priam.utils.JMXConnectorMgr;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
 import java.util.List;
 
 /**
  * Created by vinhn on 10/12/16.
  */
-public interface IClusterManagement<T> {
+public abstract class IClusterManagement<T> extends Task {
+    public enum Task {FLUSH, COMPACTION};
+    private static final Logger logger = LoggerFactory.getLogger(IClusterManagement.class);
+    private IMetricPublisher metricPublisher;
+    private Task taskType;
+    private IMeasurement measurement;
+
+    protected IClusterManagement(IConfiguration config, Task taskType, IMetricPublisher metricPublisher, IMeasurement measurement) {
+        super(config);
+        this.taskType = taskType;
+        this.metricPublisher = metricPublisher;
+        this.measurement = measurement;
+    }
+
+    @Override
+    public void execute() throws JMXConnectionException, Exception {
+        JMXConnectorMgr connMgr = null;
+        List<T> result = null;
```

Comment:
Can you simplify and introduce a `Result` entity instead of List<T>, not always I might have List. `Result` entity could be a very simple interface and calling toString method on implementations of `Result` could be used in the output that you are trying to use.

Reply:
Changed this to String as we really never use the result as such. Too much code wrapping for nothing is not a good idea. If we ever decide to use the output in any other way (other than printing) we will do this.